packages/server/integrations/TaskIntegrationManagerFactory.ts

```diff
@@ -0,0 +1,54 @@
+import {DeepPartial} from 'rethinkdb-ts/lib/internal-types'
+import {GraphQLResolveInfo} from 'graphql'
+import {IntegrationProviderServiceEnumType} from '../graphql/types/IntegrationProviderServiceEnum'
+import JiraTaskIntegrationManager from './JiraTaskIntegrationManager'
+import GitHubTaskIntegrationManager from './GitHubTaskIntegrationManager'
+import {Loaders} from '../dataloader/RootDataLoader'
+import {RValue} from '../database/stricterR'
+import Task from '../database/types/Task'
+import {AtlassianAuth} from '../postgres/queries/getAtlassianAuthByUserIdTeamId'
+import {Doc} from '../utils/convertContentStateToADF'
+import {GQLContext} from '../graphql/graphql'
+import {GitHubAuth} from '../postgres/queries/getGitHubAuthByUserIdTeamId'
+
+export type CreateTaskResponse = {
+  error?: {
+    message: string
+  }
+  integrationData?: RValue<DeepPartial<Task>>
+}
+
+export interface TaskIntegrationManager {
+  title: string
+  segmentEventName: string
+  authLoaderKey: Loaders
+
+  createTask(params: {
+    auth: AtlassianAuth | GitHubAuth
+    accessUserId: string
+    rawContentStr: string
+    projectId: string
+    createdBySomeoneElseComment?: Doc | string
+    context?: GQLContext
+    info?: GraphQLResolveInfo
+  }): Promise<CreateTaskResponse>
+
+  getCreatedBySomeoneElseComment(
+    viewerName: string,
+    assigneeName: string,
+    teamName: string,
+    teamDashboardUrl: string
+  ): Doc | string
```

Comment:
+1 Good enough for now. In the future when making more mutations generic we probably want to change this to `addCreatedBySomeoneElseComment`, but that's easier once we have the `addComment` functionality here. Let's move in small steps.

Reply:
Yeah, the current functions just accepting a comment as a param, I just did not refactor them for this task